token-lending/cli/src/main.rs

```diff
@@ -0,0 +1,700 @@
+    let source_liquidity = Token::unpack_from_slice(source_liquidity_account.data.borrow())?;
+
+    let source_liquidity_mint_account = config.rpc_client.get_account(&source_liquidity.mint)?;
+    let source_liquidity_mint =
+        Mint::unpack_from_slice(source_liquidity_mint_account.data.borrow())?;
+    let liquidity_amount = ui_amount_to_amount(ui_amount, source_liquidity_mint.decimals);
+
+    let reserve_keypair = Keypair::new();
+    let collateral_mint_keypair = Keypair::new();
+    let collateral_supply_keypair = Keypair::new();
+    let liquidity_supply_keypair = Keypair::new();
+    let liquidity_fee_receiver_keypair = Keypair::new();
+    let liquidity_host_keypair = Keypair::new();
+    let user_collateral_keypair = Keypair::new();
+    let user_transfer_authority_keypair = Keypair::new();
+
+    println!("Adding reserve {}", reserve_keypair.pubkey());
+    if config.verbose {
+        println!(
+            "Adding collateral mint {}",
+            collateral_mint_keypair.pubkey()
+        );
+        println!(
+            "Adding collateral supply {}",
+            collateral_supply_keypair.pubkey()
+        );
+        println!(
+            "Adding liquidity supply {}",
+            liquidity_supply_keypair.pubkey()
+        );
+        println!(
+            "Adding liquidity fee receiver {}",
+            liquidity_fee_receiver_keypair.pubkey()
+        );
+        println!("Adding liquidity host {}", liquidity_host_keypair.pubkey());
+        println!(
+            "Adding user collateral {}",
+            user_collateral_keypair.pubkey()
+        );
+        println!(
+            "Adding user transfer authority {}",
+            user_transfer_authority_keypair.pubkey()
+        );
+    }
+
+    let reserve_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Reserve::LEN)?;
+    let collateral_mint_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Mint::LEN)?;
+    let collateral_supply_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
+    let user_collateral_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
+    let liquidity_supply_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
+    let liquidity_fee_receiver_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
+    let liquidity_host_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
+
+    let total_balance = reserve_balance
+        + collateral_mint_balance
+        + collateral_supply_balance
+        + user_collateral_balance
+        + liquidity_supply_balance
+        + liquidity_fee_receiver_balance
+        + liquidity_host_balance;
+
+    let mut create_accounts_transaction_1 = Transaction::new_with_payer(
```

Comment:
The necessity of splitting up the transactions like this make me feel that new reserves should create accounts themselves using program derived addresses to avoid needing to stuff so many signatures into a transaction. The reason is that if account creations are in separate transactions from account initialization, anyone can front run the initialization with their own transaction to take ownership of these accounts

Reply:
I agree, it's pretty unwieldy. It also sucks when running these commands if you create them and then your `InitReserve` fails, you just have a bunch of dangling accounts. Good point about the security issue too, I wasn't aware someone else could just initialize accounts you create.

Reply:
Is it acceptable to make this change in a followup PR? It affects the program API, CLI, and tests. I'll create an issue if so.

Reply:
Yeah, follow up sounds good